lib/tokenlist.cpp

```diff
@@ -976,6 +977,23 @@ static void compileExpression(Token *&tok, AST_state& state)
         compileComma(tok, state);
 }
 
+static bool isLambdaCaptureList(Token * tok)
+{
+    // a lambda expression '[x](y){}' is compiled as:
+    // [
+    // `-(
+    //   `-{
+    // see compilePrecedence2
+    if (tok->str() != "[")
+        return false;
+    if (!tok->astOperand1() || tok->astOperand1()->str() != "(")
+        return false;
+    const Token * params = tok->astOperand1();
+    if (!params || !params->astOperand1() || params->astOperand1()->str() != "{")
+        return false;
+    return true;
+}
+
```

Comment:
Before, I had a different logic for the test inside isLambdaCaptureList but I realized that it was not completely correct.
The check now is consistent with the definition of the AST.
However, the AST might change in future to include ```x``` and ```y``` in the AST.
Furthermore a lambda without parameter ```y``` (i.e. ```[x](){}```) can also be written shorter as ```[x]{}``` (see http://en.cppreference.com/w/cpp/language/lambda (4)). Is this handled correctly?